Smoker failure due to URI::cpan

URI::cpan needs Perl >= 5.12

we are trying to use it, need to determine if it s for unit tests or not

https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true

```
Crypt::OpenSSL::Guess is up to date. (0.15)
[11](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:11)
--> Working on Test::Kwalitee
[12](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:12)
Fetching http://www.cpan.org/authors/id/E/ET/ETHER/Test-Kwalitee-1.28.tar.gz ... OK
[13](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:13)
Configuring Test-Kwalitee-1.28 ... OK
[14](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:14)
==> Found dependencies: Module::CPANTS::Analyse
[15](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:15)
--> Working on Module::CPANTS::Analyse
[16](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:16)
Fetching http://www.cpan.org/authors/id/I/IS/ISHIGAKI/Module-CPANTS-Analyse-1.01.tar.gz ... OK
[17](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:17)
Configuring Module-CPANTS-Analyse-1.01 ... OK
[18](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:18)
==> Found dependencies: Perl::PrereqScanner::NotQuiteLite
[19](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:19)
--> Working on Perl::PrereqScanner::NotQuiteLite
[20](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:20)
Fetching http://www.cpan.org/authors/id/I/IS/ISHIGAKI/Perl-PrereqScanner-NotQuiteLite-0.9914.tar.gz ... OK
[21](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:21)
Configuring Perl-PrereqScanner-NotQuiteLite-0.9914 ... OK
[22](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:22)
==> Found dependencies: URI::cpan
[23](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:23)
--> Working on URI::cpan
[24](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:24)
Fetching http://www.cpan.org/authors/id/R/RJ/RJBS/URI-cpan-1.008.tar.gz ... OK
[25](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:25)
! Configure failed for URI-cpan-1.008. See /github/home/.cpanm/work/1646941824.5444/build.log for details.
[26](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:26)
! Installing the dependencies failed: Module 'URI::cpan' is not installed
[27](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:27)
! Bailing out the installation for Perl-PrereqScanner-NotQuiteLite-0.9914.
[28](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:28)
! Installing the dependencies failed: Module 'Perl::PrereqScanner::NotQuiteLite' is not installed
[29](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:29)
! Bailing out the installation for Module-CPANTS-Analyse-1.01.
[30](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:30)
! Installing the dependencies failed: Module 'Module::CPANTS::Analyse' is not installed
[31](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:31)
! Bailing out the installation for Test-Kwalitee-1.28.
[32](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:32)
Configuring URI-cpan-1.008 ... N/A
[33](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:33)
Test::CPAN::Meta is up to date. (0.25)
[34](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:34)
Perl::MinimumVersion is up to date. (1.40)
[35](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:35)
Test::Pod::Coverage is up to date. (1.10)
[36](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:36)
Test::Pod is up to date. (1.52)
[37](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:37)
Test::MinimumVersion is up to date. (0.101082)
[38](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:38)
Crypt::OpenSSL::Bignum is up to date. (0.09)
[39](https://github.com/toddr/Crypt-OpenSSL-RSA/runs/5501686803?check_suite_focus=true#step:6:39)
Error: Process completed with exit code 1.
```

--- a/.github/workflows/linux.yml
+++ b/.github/workflows/linux.yml
@@ -9,7 +9,8 @@ on:
   pull_request:
 
 jobs:
-  perl:
+
+  ubuntu:
     env:
       PERL_USE_UNSAFE_INC: 0
       AUTHOR_TESTING: 1
@@ -18,23 +19,45 @@ jobs:
 
     runs-on: ubuntu-latest
 
+    steps:
+      - uses: actions/checkout@v2
+      - name: perl -V
+        run: perl -V
+      - name: uses install-with-cpm
+        uses: perl-actions/install-with-cpm@stable
+        with:
+          cpanfile: "cpanfile"
+      - name: Makefile.PL
+        run: perl -I$(pwd) Makefile.PL
+      - name: make test
+        run: make test
+
+  perl:
+    env:
+      PERL_USE_UNSAFE_INC: 0
+      AUTHOR_TESTING: 1
+      AUTOMATED_TESTING: 1
+      RELEASE_TESTING: 1
+
+    runs-on: ubuntu-latest
+    needs: [ubuntu]
+
     strategy:
       fail-fast: false
       matrix:
-        perl-version: [latest, 5.28, 5.26, 5.24, 5.22, "5.20", 5.18, 5.16, 5.14, 5.12 ]
+        perl-version: [latest, 5.34, 5.32, "5.30", 5.28, 5.26, 5.24, 5.22, "5.20", 5.18, 5.16, 5.14, 5.12, "5.10", 5.8 ]
 
     container:
       image: perl:${{ matrix.perl-version }}
 
     steps:
-      - uses: actions/checkout@v1
-      - name: perl -V
-        run: perl -V
+      - uses: actions/checkout@v2
+      - run: perl -V
       - name: Deps for testing
-        run: cpanm --notest Crypt::OpenSSL::Random Crypt::OpenSSL::Guess Test::Kwalitee Test::CPAN::Meta Perl::MinimumVersion Test::Pod::Coverage Test::Pod Test::MinimumVersion Crypt::OpenSSL::Bignum || /bin/true
-      - name: Deps for testing
-        run: cpanm --notest Crypt::OpenSSL::Random Crypt::OpenSSL::Guess Test::Kwalitee Test::CPAN::Meta Perl::MinimumVersion Test::Pod::Coverage Test::Pod Test::MinimumVersion Crypt::OpenSSL::Bignum      
-      - name: Makefile.PL
-        run: perl Makefile.PL
-      - name: make test
-        run: make test
+        run: |
+          cpanm --notest Crypt::OpenSSL::Random Crypt::OpenSSL::Guess Test::CPAN::Meta Perl::MinimumVersion Test::Pod::Coverage Test::Pod Test::MinimumVersion Crypt::OpenSSL::Bignum || /bin/true
+          cpanm --notest Crypt::OpenSSL::Random Crypt::OpenSSL::Guess Test::CPAN::Meta Perl::MinimumVersion Test::Pod::Coverage Test::Pod Test::MinimumVersion Crypt::OpenSSL::Bignum
+          # not available < 5.12
+          cpanm --notest Test::Kwalitee ||:
+      - run: perl Makefile.PL
+      - run: make test

